samples/23_advanced_timers.cr

```diff
@@ -0,0 +1,117 @@
+require "../src/sdl"
+require "../src/image"
+require "../src/ttf"
+
+SDL.init(SDL::Init::VIDEO); at_exit { SDL.quit }
+SDL::TTF.init; at_exit { SDL::TTF.quit }
+
+window = SDL::Window.new("SDL tutorial", 640, 480)
+renderer = SDL::Renderer.new(window, SDL::Renderer::Flags::ACCELERATED | SDL::Renderer::Flags::PRESENTVSYNC)
+
+font = SDL::TTF::Font.new(File.join(__DIR__, "data", "lazy.ttf"), 28)
+font_color = SDL::Color[0, 0, 0, 255]
+
+renderer.draw_color = SDL::Color[255, 255, 255, 255]
+bg_color = renderer.draw_color
+start_prompt_surface = font.render_shaded("Press S to Start or Stop the Timer", font_color, bg_color)
+pause_prompt_surface = font.render_shaded("Press P to Pause or Unpause the Timer", font_color, bg_color)
+
+class SampleTimer
```

Comment:
This sample timer is very nice, but can be achieved using `Time.monotonic` and doesn't need `SDL_GetTicks`. For example:

```crystal
ticks = Time.monotonic.total_milliseconds
```